core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -484,6 +484,52 @@ private boolean expandSelectItem(
     return false;
   }
 
+  private static SqlNode expandCommonColumn(SqlSelect sqlSelect,
+      SqlNode selectItem, SelectScope scope) {
+    if (!(selectItem instanceof SqlIdentifier)) {
+      return selectItem;
+    }
+
+    final SqlIdentifier identifier = (SqlIdentifier) selectItem;
+    if (!identifier.isSimple()) {
+      return selectItem;
+    }
+
+    final SqlNode from = sqlSelect.getFrom();
+    if (from == null || !(from instanceof SqlJoin)) {
+      return selectItem;
+    }
+
+    final SqlJoin join = (SqlJoin) from;
+    if (join.getConditionType() != JoinConditionType.USING) {
```

Comment:
If our target is `USING` clause, I'm not sure if it deserve to add a test for below sql
```
SELECT deptno, name FROM emp JOIN dept on emp.deptno = dept.deptno
```
and assert that it will throw exception `Column 'DEPTNO' is ambiguous`

Reply:
I will check whether there is already a test like you propose. If not, I will add one.